mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -1927,9 +1927,9 @@ static std::optional<int64_t> getMemoryFootprintBytes(Block &block,
       return opInst->emitError("error obtaining memory region\n");
     }
 
-    auto it = regions.find(region->memref);
-    if (it == regions.end()) {
-      regions[region->memref] = std::move(region);
+    auto [it, inserted] = regions.try_emplace(region->memref);
+    if (inserted) {
+      it->second = std::move(region);
```

Comment:
Technically you could also do the std::move in try_emplace (because it will only actually get moved if inserted), but it's probably more confusing than this variant...

Reply:
Yeah, it's safe but confusing.  Personally, when I see `std::move`, I don't want to see any use until the end of the scope.